pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -585,6 +585,14 @@ func translateRouteMatch(in *networking.HTTPMatchRequest, node *model.Proxy) *ro
 		out.Headers = append(out.Headers, &matcher)
 	}
 
+	if util.IsIstioVersionGE14(node) {
+		for name, stringMatch := range in.WithoutHeaders {
```

Comment:
this is not what I meant. You dont need a version guard around this code as this functionality is available in all recent data plane envoys. I was suggesting to change the translateHeadermatch code to use envoy's safe_regex always and not use the regex. This code is okay as is, but the version check is redundant.